GameOfLife.java

```diff
@@ -58,21 +55,70 @@ private static void play(String fileName) {
 	// and the rightmost columns. Thus the actual board is surrounded by a "frame" of zeros. You can think
 	// of this frame as representing the infinite number of dead cells that exist in every direction.
 	// This function assumes that the input file contains valid data, and does no input testing.
-	private static int[][] read(String fileName) {
+	public static int[][] read(String fileName) {
 		In in = new In(fileName); // Constructs an In object for reading the input file
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
-		//// Replace the following statement with your code.
-		return null;
+		for(int i=0;i<board.length;i++)//setting zero all over the board
+			for(int j=0;j<board[0].length;j++)
+				board[i][j]=0;
+		int countline = 1;
+		String cells = in.readLine();
+		while(true)
+		{
+			
+				if(cells.length()==0)
+				{
+					countline++;
+				}
+				else break;
+			cells = in.readLine();				
+		}
```

Comment:
Why did you use `while(true)` then break it, instead clarifying when the loop should be broken.
```suggestion
		while(cells.length()==0)
		{
			countline++;
			cells = in.readLine();				
		}
```